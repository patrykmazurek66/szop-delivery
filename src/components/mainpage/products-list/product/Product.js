import "./product.css";
import { Slide, Snackbar, Button, Modal } from "@mui/material";
import { useState } from "react";
import ProductModal from "../../product-modal/ProductModal";
import addToCart from "./AddToCart";

export default function Product({ name, imagePath, description, price, rating, nutrition }) {
  const [open, setOpen] = useState(false);

  const handleClick = () => {
    setOpen(true);
    addToCart({ name, price, quantity: 1 });
  };

  const handleClose = (event, reason) => {
    if (reason === "clickaway") {
      return;
    }
    setOpen(false);
  };

  const [openModal, setOpenModal] = useState(false);
  const handleOpenModal = () => setOpenModal(true);
  const handleCloseModal = () => setOpenModal(false);

  return (
    <div className="product-container">
      <Button onClick={handleOpenModal} sx={{ padding: 0, border: "none" }}>
        <img src={"/food/thumbnails/" + imagePath} alt={name} className="product--img" />
      </Button>

      <div className="product--right-col">
        <div className="product--name-desc">
          <Button variant="text" onClick={handleOpenModal} className="product--name-btn">
            <p className="product-name">{name}</p>
          </Button>
          <p className="product-desc">{description}</p>
        </div>
        <div className="product--price-btn">
          <p className="product-price">{price}$</p>
          <Button variant="contained" onClick={handleClick}>
            ADD +
          </Button>
        </div>
      </div>
      <Snackbar
        autoHideDuration={4000}
        open={open}
        onClose={handleClose}
        TransitionComponent={function TransitionLeft(props) {
          return <Slide {...props} direction="right" />;
        }}
        message={name + " has been added to your cart!"}
        key={name}
      />
      <Modal
        open={openModal}
        onClose={handleCloseModal}
        aria-labelledby="modal-modal-title"
        aria-describedby="modal-modal-description"
      >
        <div>
          <ProductModal
            name={name}
            imgPath={imagePath}
            description={description}
            price={price}
            rating={rating}
            nutrition={nutrition}
          />
        </div>
      </Modal>
    </div>
  );
}
