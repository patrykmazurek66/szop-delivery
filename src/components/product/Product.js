import "./product.css";
import Button from "../button/Button";

export default function Product({ name, imagePath, description, price }) {
  const addToCart = item => {
    localStorage.setItem("cart", item);
  };

  return (
    <div className="product-container">
      <img src={imagePath} alt={name} className="product--img" />
      <div className="product--right-col">
        <div className="product--name-desc">
          <p className="product-name">{name}</p>
          <p className="product-desc">{description}</p>
        </div>
        <div className="product--price-btn">
          <p className="product-price">{price}$</p>
          <Button small primary onClick={() => addToCart(name)}>
            ADD +
          </Button>
        </div>
      </div>
    </div>
  );
}
