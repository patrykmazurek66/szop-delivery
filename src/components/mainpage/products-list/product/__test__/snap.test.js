import renderer from "react-test-renderer";
import Product from "../Product";

it("Product renders correctly", () => {
  const tree = renderer
    .create(
      <Product
        name={"nazwa"}
        description={"item.description"}
        price={"item.price"}
        key={"item.id"}
        imagePath={"blank.png"}
        rating={5}
        nutrition={{
          calories: 520,
          fat: 5,
          proteins: 20,
          carbohydrates: 70,
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
