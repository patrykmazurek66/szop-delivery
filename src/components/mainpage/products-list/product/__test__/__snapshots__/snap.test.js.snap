// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product renders correctly 1`] = `
<div
  className="product-container"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-ipyyx1-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <img
      alt="nazwa"
      className="product--img"
      src="/food/thumbnails/blank.png"
    />
  </button>
  <div
    className="product--right-col"
  >
    <div
      className="product--name-desc"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium product--name-btn css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <p
          className="product-name"
        >
          nazwa
        </p>
      </button>
      <p
        className="product-desc"
      >
        item.description
      </p>
    </div>
    <div
      className="product--price-btn"
    >
      <p
        className="product-price"
      >
        item.price
        $
      </p>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        ADD +
      </button>
    </div>
  </div>
</div>
`;
